Make mbsync not eat letters from last argument of one-line options.
A fix for #425838. -- Lubomir Kundrak <lkundrak@redhat.com>

diff -urp isync-1.0.3/src/main.c isync-1.0.3.args/src/main.c
--- isync-1.0.3/src/main.c	2006-11-01 07:54:31.000000000 +0100
+++ isync-1.0.3.args/src/main.c	2007-12-16 19:55:13.000000000 +0100
@@ -178,7 +178,7 @@ main( int argc, char **argv )
 	}
 	arc4_init();
 
-	for (oind = 1, ochar = 0; oind < argc; ) {
+	for (oind = 1, ochar = 0; oind < argc || (ochar && *ochar); ) {
 		if (!ochar || !*ochar) {
 			if (argv[oind][0] != '-')
 				break;
